node/service/src/rpc/frequency_rpc.rs

```diff
@@ -11,43 +11,130 @@ use common_primitives::rpc::RpcEvent;
+	pub fn new(client: Arc<C>, pool: Arc<P>) -> Self {
+		Self { client, pool, _marker: Default::default() }
 	}
 }
 
 #[async_trait]
-impl<C, Block> FrequencyRpcApiServer<Block> for FrequencyRpcHandler<C, Block>
+impl<P, C, Block, AccountId, Nonce> FrequencyRpcApiServer<Block, AccountId, Nonce>
+	for FrequencyRpcHandler<P, C, Block>
 where
 	Block: BlockT,
+	C: HeaderBackend<Block>,
 	C: Send + Sync + 'static,
 	C: ProvideRuntimeApi<Block>,
 	C::Api: AdditionalRuntimeApi<Block>,
+	C::Api: AccountNonceApi<Block, AccountId, Nonce>,
+	P: TransactionPool + 'static,
+	AccountId: Clone + Codec,
+	Nonce: Clone + Encode + Decode + AtLeast32Bit + 'static,
 {
 	fn get_events(&self, at: <Block as BlockT>::Hash) -> RpcResult<Vec<RpcEvent>> {
 		let api = self.client.runtime_api();
 		return map_rpc_result(api.get_events(at))
 	}
+
+	fn get_missing_nonce_values(&self, account: AccountId) -> RpcResult<Vec<Nonce>> {
+		let api = self.client.runtime_api();
+		let best = self.client.info().best_hash;
+
+		let nonce = api.account_nonce(best, account.clone()).map_err(|e| {
+			CallError::Custom(ErrorObject::owned(1, "Unable to query nonce.", Some(e.to_string())))
+		})?;
+		Ok(get_missing_nonces(&*self.pool, account, nonce))
+	}
+}
+
+/// Finds any missing nonce values inside Future pool and return them as result
+fn get_missing_nonces<P, AccountId, Nonce>(pool: &P, account: AccountId, nonce: Nonce) -> Vec<Nonce>
+where
+	P: TransactionPool,
+	AccountId: Clone + Encode,
+	Nonce: Clone + Encode + Decode + AtLeast32Bit + 'static,
+{
+	// Now we need to query the transaction pool
+	// and find transactions originating from the same sender.
+	// Since extrinsics are opaque to us, we look for them using
+	// `provides` tag. And increment the nonce if we find a transaction
+	// that matches the current one.
+	let mut current_nonce = nonce.clone();
+	let encoded_account = account.clone().encode();
+	let mut current_tag = (account.clone(), nonce).encode();
+	for tx in pool.ready() {
+		// since transactions in `ready()` need to be ordered by nonce
+		// it's fine to continue with current iterator.
+		if tx.provides().get(0) == Some(&current_tag) {
+			current_nonce += One::one();
+			current_tag = (account.clone(), current_nonce.clone()).encode();
+		}
+	}
+
+	let mut result = vec![];
+	let mut my_in_future: Vec<_> = pool
+		.futures()
+		.into_iter()
+		.filter_map(|x| match x.provides().get(0) {
+			// filtering transactions by account
+			Some(tag) if tag.starts_with(&encoded_account) => {
+				if let Ok(nonce) = Nonce::decode(&mut &tag[encoded_account.len()..]) {
+					return Some(nonce)
+				}
+				None
+			},
+			_ => None,
+		})
+		.collect();
+	my_in_future.sort();
+
+	for future_nonce in my_in_future {
+		while current_nonce < future_nonce {
+			result.push(current_nonce.clone());
+			current_nonce += One::one();
+
+			// short circuit if we reached the limit
+			if result.len() == MAX_RETURNED_MISSING_NONCE_SIZE {
+				return result
+			}
```

Comment:
code golf: any benefit to using `BoundedVec` for the result array, instead of this check?

Reply:
Even though I could use it I didn't since I didn't want the result to error out if it was out of bounds and I just wanted to restrict the result size without having extra complexity.